scripts/requirements-build-test.txt

```diff
@@ -22,3 +22,6 @@ mypy
 
 # used for mocking functions in pytest
 mock>=4.0.1
+
+# used in MCUboot for signing and key management
+imgtool>=1.6.0
```

Comment:
Not sure if the answer is yes or no, but can you check if cmake/mcuboot.cmake support for finding imgtool via PATH or from mcuboot/scripts should be extracted instead of making this a generic requirement?

Reply:
The problem is that MCUboot in this case, @mbolivar-nordic , is built via TF-M, which is an external project. It is not build with Zephyr.

Reply:
...in other words,  cmake/mcuboot.cmake, is not processed at all.

Reply:
> ...in other words, cmake/mcuboot.cmake, is not processed at all.

Yes, of course, I'm asking if factoring out that code to find imgtool into shared code would be a better approach. That is why I said "extracted".

Reply:
By putting this here it looks like you are basically forcing everybody who uses zephyr+mcuboot+signing via the build system to get imgtool from pip instead of from the mcuboot repository. When I added mcuboot.cmake, it was a deliberate choice to favor imgtool installed via pip over the one found in the mcuboot repository because imgtool was **not** installed by default in the zephyr requirements.txt. This is changing that default.

Reply:
OK, so to conclude, what would be your suggestion @mbolivar-nordic ?

I cannot, really, hack the external project to look for imgtool from anywhere else than PIP; that will be a nightmare to maintain. I am OK with removing it from requirements.txt, since I already added a doc entry in the TFM samples regarding the python requirements for building TFM binaries (FYI @microbuilder ) so maybe we can afford to skip adding this here...

No strong opinion, but I would also like @microbuilder to comment.

Reply:
> OK, so to conclude, what would be your suggestion @mbolivar-nordic ?

My suggestion is to find imgtool in host-tools.cmake by extracting the code in mcuboot.cmake and setting an IMGTOOL CMake variable to something you can run to run imgtool.

My point is that if imgtool is a host tool that is needed in multiple places now (as this PR seems to be making the case), it should probably be treated like any other host tool in the sense of its discovery getting moved to host-tools.cmake. Previously I didn't do this because mcuboot.cmake was a special purpose use case.

cc @tejlmand

Reply:
@mbolivar-nordic based on @microbuilder 's comment below, we 'll remove this edit in the requirements-build-test.txt.

Reply:
> we 'll remove this edit in the requirements-build-test.txt.

Thanks.